test/org/zaproxy/zap/extension/fuzz/payloads/generator/JsonPayloadGeneratorTest.java

```diff
@@ -0,0 +1,96 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.fuzz.payloads.generator;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import org.junit.Before;
+import org.junit.Test;
+import org.zaproxy.zap.extension.fuzz.payloads.DefaultPayload;
+import org.zaproxy.zap.utils.ResettableAutoCloseableIterator;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+public class JsonPayloadGeneratorTest {
+
+    @Before
+    public void setUp() {
+
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldRejectMissionJsonBase() {
+        new JsonPayloadGenerator(null, 1);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldRejectInvalidJsonBase() {
+        new JsonPayloadGenerator("'an invalid value", 1);
+    }
+
+    @Test
+    public void shouldHaveAtLeastOnePayload() {
+        String originalJson = "{\"x\": \"hello\"}";
+        JsonPayloadGenerator generator = new JsonPayloadGenerator(originalJson, 1);
+        assertThat(generator.iterator().hasNext(), is(true));
+    }
+
+    @Test
+    public void shouldGenerateOneFuzzedPayload() {
+        String originalJson = "{\"x\": \"hello\"}";
+        JsonPayloadGenerator generator = new JsonPayloadGenerator(originalJson, 1);
+        DefaultPayload next = generator.iterator().next();
+        JsonElement json = toGson(next.getValue());
+        assertThat(json, is(not(toGson(originalJson))));
+    }
+
+    @Test
+    public void shouldNotGenerateTooManyPayloads() {
+        String originalJson = "{\"x\": \"hello\"}";
+        JsonPayloadGenerator generator = new JsonPayloadGenerator(originalJson, 1);
+        ResettableAutoCloseableIterator<DefaultPayload> iterator = generator.iterator();
+        iterator.next();
+        assertThat(iterator.hasNext(), is(false));
+    }
+
+    @Test
+    public void shouldGenerateMultipleMutants() {
+        String originalJson = "{\"x\": \"hello\"}";
+        JsonPayloadGenerator generator = new JsonPayloadGenerator(originalJson, 20);
+        ResettableAutoCloseableIterator<DefaultPayload> iterator = generator.iterator();
+        Set<String> mutants = new HashSet<>();
+        while (iterator.hasNext()) {
+            mutants.add(iterator.next().toString());
+        }
+        // Don't generate an infinite number of mutants
+        assertThat(mutants.size(), greaterThan(15));
```

Comment:
Why not assert that it generated the expected number?

Reply:
Yes, I can
I wanted to also highly the fact that, even if you request 100 mutants, the library will generate a finite number of mutants.
I will add a comment to explain this point.

Reply:
Ah, OK, didn't get that. Sounds good.